anvil-python/anvil/jobs/juju.py

```diff
@@ -12,5 +12,50 @@
 # implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+import logging
+import subprocess
+
+from rich.status import Status
+from sunbeam.commands.juju import ScaleJujuStep
+from sunbeam.jobs.common import Result, ResultType
 
 CONTROLLER = "anvil-controller"
+LOG = logging.getLogger(__name__)
+
+
+class AnvilScaleJujuStep(ScaleJujuStep):
```

Comment:
As far as I can see we are using the exact same Step as Sunbeam. Is there a reason for re-implementing the class?

Reply:
In addition, inside the manifest we have the `scale_args`. This is something that we need to consume and append in the enable-ha command. I can see that you are using the extra_args to add `--to` option. What about letting the user decide what to do with Juju HA? Or, if we want to have this logic when joining new nodes, let's continue using --to and remove scale_args from JujuManifest by implementing a new Pydantic class JujuManifest and use this instead. Warning: This one can be a tricky change.

Reply:
It's not quite exactly the same class; the model name in the `juju wait-for` command is slightly different. When just using `ScaleJujuStep` as is, this command hangs until the timeout even though the controllers are ready.

For the `scale_args` in the manifest, I don't know why someone _wouldn't_ want Juju HA, but if you think it really is necessary I can add it. Also, as the `--to` args changes based on how many machines have already joined, is that something we can take from the manifest? 

I'm going to go with the latter option of removing `scale_args` in a new `JujuManifest` Pydantic class for now

Reply:
Thank you for taking care of the JujuManifest class. As far as ScaleJujuStep is concerned, it seems that the sunbeam version cannot work for us. You were right that the wait-for is different but can you spot the difference? It is waiting on the `-m controller`. Based on its [documentation](https://juju.is/docs/juju/juju-wait-for-application) `-m` stands for model. In our case the model name (not short-name) is `admin/controller`. It can be observed if you run `juju show-model`. All the above would work for us but not the sunbeam one:

- `juju wait-for application -m anvil-controller:admin/controller controller`
- `juju wait-for application -m c88fac18-5d7a-42f2-8747-bcf7ecea6270 controller`
- `juju wait-for application -m admin/controller controller`

Based on all of the above feel free to remove the import and the inheritance from sunbeam